api4/drafts.go

```diff
@@ -120,11 +120,16 @@ func deleteDraft(c *Context, w http.ResponseWriter, r *http.Request) {
 	channelID := c.Params.ChannelId
 
 	draft, err := c.App.GetDraft(userID, channelID, rootID)
-	if err != nil || c.AppContext.Session().UserId != draft.UserId {
+	if c.AppContext.Session().UserId != draft.UserId {
 		c.SetPermissionError(model.PermissionDeletePost)
 		return
 	}
 
+	if err != nil {
```

Comment:
Err check will come first then this `c.AppContext.Session().UserId != draft.UserId`. Otherwise draft could be nil, in case of error.